src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -2694,32 +2709,38 @@ namespace HVACUnitarySystem {
                         UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(DesignSpecMSHP(MSHPIndex).NumOfSpeedCooling);
                 }
                 UnitarySystem(UnitarySysNum).IdleVolumeAirRate =
-                    UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(1) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+                    UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
                 UnitarySystem(UnitarySysNum).IdleMassFlowRate =
-                    UnitarySystem(UnitarySysNum).CoolMassFlowRate(1) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+                    UnitarySystem(UnitarySysNum).CoolMassFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
                 UnitarySystem(UnitarySysNum).IdleSpeedRatio =
-                    UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(1) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+                    UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+            } else {
+                UnitarySystem(UnitarySysNum).IdleVolumeAirRate = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow;
+                UnitarySystem(UnitarySysNum).IdleMassFlowRate = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirMassFlow;
+                UnitarySystem(UnitarySysNum).IdleSpeedRatio = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow / UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds);
```

Comment:
2 things here.  `NumSpeeds` could be zero at this point (same at line 2682 above) - might be what's causing the segfault in the CI tests.  Second, while `UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds)` will typically be the full cooling flow (because we expect the top speed to have a flow fraction of 1.0) , it won't necessarily be that.  The denominator here should be whatever IdleSpeedRatio gets multiplied by later - is it the cooling or heating flow or the max or min of these?

Reply:
The speed ratio is passed to the OnOff fan model. So it has to be the ratio of flow to max flow of this operating mode. Pretty sure I added speed ratio a long time ago. 

    //      adjust RTF to be in line with speed ratio (i.e., MaxAirMassFlowRate is not MAX when SpeedRatio /= 1)
    //      PLR = Mdot/MAXFlow => Mdot/(MAXFlow * SpeedRatio), RTF = PLR/PLF => PLR/SpeedRatio/PLF = RTF / SpeedRatio
    if (SpeedRatio > 0.0) Fan(FanNum).FanRuntimeFraction = min(1.0, Fan(FanNum).FanRuntimeFraction / SpeedRatio);
    SpeedRaisedToPower = CurveValue(Fan(FanNum).FanPowerRatAtSpeedRatCurveIndex, SpeedRatio);
    Fan(FanNum).FanPower *= SpeedRaisedToPower / EffRatioAtSpeedRatio;

FanSystemModel also uses this:

    if (present(flowFraction)) {
        localFlowFraction = flowFraction;
        localAirMassFlow[0] = localFlowFraction * m_maxAirMassFlowRate;

So it looks to me as though speed ratio is always with respect to the max fan flow.

Reply:
I'm adding this in InitLoadBasedControl to get this right and should overwrite what is done during sizing. The speed ratio should be relative to max fan flow but I may not know the fan flow during sizing. I'll have to think more about what to do with set point based control, since that is constant fan, but it seems to me these should be initialized the same way regardless of control type. Once I know the fan flow rate I'll initialized these variables:

    if (MultiOrVarSpeedHeatCoil(UnitarySysNum) || MultiOrVarSpeedCoolCoil(UnitarySysNum)) {
        if (MultiOrVarSpeedCoolCoil(UnitarySysNum)) {
            int NumSpeeds = UnitarySystem(UnitarySysNum).NumOfSpeedCooling;
            if (!allocated(UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio)) UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio.allocate(NumSpeeds);
            for (int Iter = 1; Iter <= NumSpeeds; ++Iter) {
                UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(Iter) = UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) / UnitarySystem(UnitarySysNum).ActualFanVolFlowRate;
            }
        }
        if (MultiOrVarSpeedHeatCoil(UnitarySysNum)) {
            int NumSpeeds = UnitarySystem(UnitarySysNum).NumOfSpeedHeating;
            if (!allocated(UnitarySystem(UnitarySysNum).MSHeatingSpeedRatio)) UnitarySystem(UnitarySysNum).MSHeatingSpeedRatio.allocate(NumSpeeds);
            for (int Iter = 1; Iter <= NumSpeeds; ++Iter) {
                UnitarySystem(UnitarySysNum).MSHeatingSpeedRatio(Iter) = UnitarySystem(UnitarySysNum).HeatVolumeFlowRate(Iter) / UnitarySystem(UnitarySysNum).ActualFanVolFlowRate;
            }
        }
        UnitarySystem(UnitarySysNum).IdleSpeedRatio = UnitarySystem(UnitarySysNum).IdleVolumeAirRate / UnitarySystem(UnitarySysNum).ActualFanVolFlowRate;
    }

I wonder if I should use MaxNoCoolHeatAirVolFlow on the last line instead of IdleVolumeAirRate?